packages/cozy-procedures/src/Procedure.jsx

```diff
@@ -1,20 +1,59 @@
 import React from 'react'
+import PropTypes from 'prop-types'
 import get from 'lodash/get'
 import { withClient } from 'cozy-client'
-
+import { Document } from 'cozy-doctypes'
+import { Spinner } from 'cozy-ui/transpiled/react/'
 import creditApplicationTemplate from './templates/creditApplicationTemplate'
-
 class Procedure extends React.Component {
   componentDidMount() {
-    this.props.initPersonalData(
+    const {
+      initPersonalData,
+      initDocuments,
+      fetchMyself,
+      fetchDocument,
+      client,
+      setStatusDemarche
+    } = this.props
+    //We init our Document model here to be able to use CozyFile or AdministrativeProcedure models where we want
+    if (!Document.cozyClient) {
+      Document.registerClient(client)
+    }
+    initPersonalData(
       get(creditApplicationTemplate, 'personalData.schema.properties', {})
     )
-    this.props.fetchMyself(this.props.client)
+
+    initDocuments(get(creditApplicationTemplate, 'documents'))
+    //Since init is done, we tell the app we can start to render thing
+    setStatusDemarche({ initiated: true })
+    fetchMyself(client)
+
+    const { documents: documentsTemplate } = creditApplicationTemplate
+    Object.keys(documentsTemplate).map(document => {
+      fetchDocument(client, document)
+    })
   }
 
   render() {
-    return <div className="u-p-1 u-maw-6">{this.props.children}</div>
+    return (
+      <div className="u-p-1 u-maw-6">
+        {this.props.initiated === true ? (
+          this.props.children
+        ) : (
+          <Spinner size="xxlarge" />
```

Comment:
I understand it avoids unnecessary renders but are we want to block the whole UI until we haven't fetch everything? (the user could see what are the sections)

Reply:
We don't block until we have fetched, we just block until we have initiated our app.